cmd/internal/cli/actions.go

```diff
@@ -185,6 +184,33 @@ func replaceURIWithImage(cmd *cobra.Command, args []string) {
 	return
 }
 
+// TODO: Let's stick this in another file so that that CLI is just CLI
```

Comment:
You wouldn't have access to the `IsSyOS` or `VM` variables if you moved it outside the CLI package would you?

Reply:
Well two things here:

1. Technically you're correct here, you will lose access to package-wide variables when moving a function outside of that package. However, in its current state, `vmStarter` takes `cmd *cobra.Command` as an argument which *does* give access to that flag's value.
2. We probably want to move this function to `internal/app/singularity`, and remove `cmd *cobra.Command` as an argument. Ideally, we want to parse the CLI options which have been set in `cmd/internal/cli` into some well-defined set of arguments (see the capability code for an example of this design pattern).

Reply:
What I was thinking was creating a struct with the configuration from the command line, and passing that to a command that needs access to to that information. So, instead of having a bunch of global variables, there would be a single one that could be passed around as needed.

Reply:
@mem I think that's a great goal to address in another PR at some point down the line